packages/signer/src/signer.ts

```diff
@@ -107,12 +90,12 @@ class Signer {
     // remove recoveryParam from s first byte
     s[0] &= 0x7f;
 
-    const publicKey = getCurve()
-      .recoverPubKey(getBytesCopy(data), { r, s }, recoveryParam)
-      .encode('array', false)
-      .slice(1);
+    const sig = new secp256k1.Signature(BigInt(hexlify(r)), BigInt(hexlify(s))).addRecoveryBit(
```

Comment:
Interesting seeing another package opting for `BigInt` - #1542

Reply:
Good catch! Thanks for linking it to the issue.